services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -175,8 +176,13 @@ private void startServices() {
   }
 
   private SafeFuture<?> initialize() {
+    final StoreOptions storeOptions =
```

Comment:
I'm still in two minds but I wonder if we should just make `StoreOptions` an interface that `TekuConfiguration` implements, then we don't have to rebuild a different config object, but still narrow down what gets passed in.  It already does it for `MetricsConfig` which I think was mostly driven by module dependencies but it seems to work (and eventually `TekuConfiguration` could just hold multiple separate objects so you `getMetricsConfig` out of it instead of using it directly.

Happy to go either way with this at the moment though.

Reply:
I don't think that works because it would create circular dependencies:  `storage` depends on `util`, so `util` can't depend on `storage`.

Reply:
Ugh.  Yeah we'd probably have to define `StoreConfig` in util.  Not worth the fight now I don't think.